From b533ab1c24e6385736a68dd6a4b21ccb05e21781 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 23 Oct 2024 10:35:43 +0200
Subject: [PATCH 1466/1695] plat-stm32mp1: remove unused
 stm32mp_nsec_can_access_clock()

Remove unused platform function stm32mp_nsec_can_access_clock().

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b7ec7447ad22a31ea462cbec4005c729bc090721 plat-stm32mp1: remove unused stm32mp_nsec_can_access_clock()]
Change-Id: If5d921ae86a9a61aa8056f089738ba0f58f081c2
Link: https://github.com/OP-TEE/optee_os/commit/ded57353b0fc6f5221983f668dc0daaef4d7f90f
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/413525
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../arch/arm/plat-stm32mp1/shared_resources.c | 69 -------------------
 core/arch/arm/plat-stm32mp1/stm32_util.h      | 11 ---
 2 files changed, 80 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/shared_resources.c b/core/arch/arm/plat-stm32mp1/shared_resources.c
index f73cbac55..3a6e8be1d 100644
--- a/core/arch/arm/plat-stm32mp1/shared_resources.c
+++ b/core/arch/arm/plat-stm32mp1/shared_resources.c
@@ -472,75 +472,6 @@ bool stm32mp_gpio_bank_is_secure(unsigned int bank)
 	return secure > 0 && secure == get_gpioz_nbpin();
 }
 
-bool stm32mp_nsec_can_access_clock(unsigned long clock_id)
-{
-	enum stm32mp_shres shres_id = STM32MP1_SHRES_COUNT;
-
-	/* Oscillators and PLLs are visible from non-secure world */
-	COMPILE_TIME_ASSERT(CK_HSE == 0 &&
-			    (CK_HSE + 1) == CK_CSI &&
-			    (CK_HSE + 2) == CK_LSI &&
-			    (CK_HSE + 3) == CK_LSE &&
-			    (CK_HSE + 4) == CK_HSI &&
-			    (CK_HSE + 5) == CK_HSE_DIV2 &&
-			    (PLL1_P + 1) == PLL1_Q &&
-			    (PLL1_P + 2) == PLL1_R &&
-			    (PLL1_P + 3) == PLL2_P &&
-			    (PLL1_P + 4) == PLL2_Q &&
-			    (PLL1_P + 5) == PLL2_R &&
-			    (PLL1_P + 6) == PLL3_P &&
-			    (PLL1_P + 7) == PLL3_Q &&
-			    (PLL1_P + 8) == PLL3_R);
-
-	if (clock_id <= CK_HSE_DIV2 ||
-	    (clock_id >= PLL1_P && clock_id <= PLL3_R))
-		return true;
-
-	switch (clock_id) {
-	case RTCAPB:
-	case CK_MPU:
-	case CK_AXI:
-	case BSEC:
-		return true;
-	case GPIOZ:
-		return !stm32mp_gpio_bank_is_secure(GPIO_BANK_Z);
-	case SPI6_K:
-		shres_id = STM32MP1_SHRES_SPI6;
-		break;
-	case I2C4_K:
-		shres_id = STM32MP1_SHRES_I2C4;
-		break;
-	case I2C6_K:
-		shres_id = STM32MP1_SHRES_I2C6;
-		break;
-	case USART1_K:
-		shres_id = STM32MP1_SHRES_USART1;
-		break;
-	case IWDG1:
-		shres_id = STM32MP1_SHRES_IWDG1;
-		break;
-	case CRYP1:
-		shres_id = STM32MP1_SHRES_CRYP1;
-		break;
-	case HASH1:
-		shres_id = STM32MP1_SHRES_HASH1;
-		break;
-	case RNG1_K:
-		shres_id = STM32MP1_SHRES_RNG1;
-		break;
-	case RTC:
-		shres_id = STM32MP1_SHRES_RTC;
-		break;
-	case CK_MCU:
-		shres_id = STM32MP1_SHRES_MCU;
-		break;
-	default:
-		return false;
-	}
-
-	return !stm32mp_periph_is_secure(shres_id);
-}
-
 bool stm32mp_nsec_can_access_reset(unsigned int reset_id)
 {
 	enum stm32mp_shres shres_id = STM32MP1_SHRES_COUNT;
diff --git a/core/arch/arm/plat-stm32mp1/stm32_util.h b/core/arch/arm/plat-stm32mp1/stm32_util.h
index 65ad71580..448322344 100644
--- a/core/arch/arm/plat-stm32mp1/stm32_util.h
+++ b/core/arch/arm/plat-stm32mp1/stm32_util.h
@@ -84,17 +84,6 @@ static inline void stm32mp_register_online_cpu(void)
 uint32_t may_spin_lock(unsigned int *lock);
 void may_spin_unlock(unsigned int *lock, uint32_t exceptions);
 
-#ifdef CFG_STM32MP1_SHARED_RESOURCES
-/* Return true if @clock_id is shared by secure and non-secure worlds */
-bool stm32mp_nsec_can_access_clock(unsigned long clock_id);
-#else /* CFG_STM32MP1_SHARED_RESOURCES */
-static inline bool stm32mp_nsec_can_access_clock(unsigned long clock_id
-						 __unused)
-{
-	return true;
-}
-#endif /* CFG_STM32MP1_SHARED_RESOURCES */
-
 #ifdef CFG_STM32MP1_SHARED_RESOURCES
 /* Return true if and only if @reset_id relates to a non-secure peripheral */
 bool stm32mp_nsec_can_access_reset(unsigned int reset_id);
-- 
2.47.2

